include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -109,16 +127,48 @@ class MjbotsControlLoop : public AbstractRealtimeObject {
 
 /******************************************Implementation**************************************************************/
 
-template<class log_type, class input_type>
-MjbotsControlLoop<log_type, input_type>::MjbotsControlLoop(std::vector<kodlab::mjbots::JointMoteus> joints, const ControlLoopOptions &options) :
-    AbstractRealtimeObject(options.realtime_params.main_rtp, options.realtime_params.can_cpu),
+template<class log_type, class input_type, class robot_type>
+MjbotsControlLoop<log_type, input_type, robot_type>::MjbotsControlLoop(std::vector<kodlab::mjbots::JointMoteus> joints, const ControlLoopOptions &options)
+  : MjbotsControlLoop<log_type, input_type,robot_type>( make_share_vector(joints), options){}
+
+template<class log_type, class input_type, class robot_type>
+MjbotsControlLoop<log_type, input_type, robot_type>::MjbotsControlLoop(std::vector<std::shared_ptr<kodlab::mjbots::JointMoteus>> joint_ptrs, const ControlLoopOptions &options)
+  : AbstractRealtimeObject(options.realtime_params.main_rtp, options.realtime_params.can_cpu),
+    lcm_sub_(options.realtime_params.lcm_rtp, options.realtime_params.lcm_cpu, options.input_channel_name) 
+{
+  robot_ = std::make_shared<robot_type>(  joint_ptrs,
+                                          options.soft_start_duration,
+                                          options.max_torque);
+  
+  mjbots_interface_ = std::make_shared<kodlab::mjbots::MjbotsRobotInterface>( joint_ptrs,
+                                          options.realtime_params,
+                                          options.imu_mounting_deg,
+                                          options.attitude_rate_hz);
+  num_motors_ = robot_->joints.size();
+  SetupOptions(options);
+}
+
+template<class log_type, class input_type, class robot_type>
+MjbotsControlLoop<log_type, input_type, robot_type>::MjbotsControlLoop(std::shared_ptr<robot_type>robot_in, const ControlLoopOptions &options)
+  : AbstractRealtimeObject(options.realtime_params.main_rtp, options.realtime_params.can_cpu),
     lcm_sub_(options.realtime_params.lcm_rtp, options.realtime_params.lcm_cpu, options.input_channel_name) {
+  // Create robot object
+  robot_ = robot_in;
+  mjbots_interface_ = std::make_shared<kodlab::mjbots::MjbotsRobotInterface>( robot_->joints,
```

Comment:
This composition one of the important changes along with the additions of a RobotInterface class

Reply:
Everyone seems on board with this.